pkg/aws/aws.go

```diff
@@ -68,25 +82,77 @@ func durationFromString(config map[string]string) (time.Duration, error) {
 	return time.ParseDuration(d)
 }
 
-func authenticateAWSCredentials(ctx context.Context, config map[string]string, assumeRoleDuration time.Duration) (*credentials.Credentials, string, error) {
-	var creds *credentials.Credentials
-	var assumedRole string
-
-	switch {
-	case config[AccessKeyID] != "" && config[SecretAccessKey] != "":
-		// If AccessKeys were provided - use those
-		creds = credentials.NewStaticCredentials(config[AccessKeyID], config[SecretAccessKey], "")
-	case os.Getenv(webIdentityTokenFilePathEnvKey) != "" && os.Getenv(roleARNEnvKey) != "":
-		sess, err := session.NewSessionWithOptions(session.Options{AssumeRoleDuration: assumeRoleDuration})
-		if err != nil {
-			return nil, "", errors.Wrap(err, "Failed to create session to initialize Web Identify credentials")
-		}
-		creds = getCredentialsWithDuration(sess, assumeRoleDuration)
+func authenticateAWSCredentials(
+	config map[string]string,
+	assumeRoleDuration time.Duration,
+) (creds *credentials.Credentials, assumedRole string, err error) {
+	// If AccessKeys were provided - use those
+	if creds = fetchStaticAWSCredentials(config); creds != nil {
+		return
+	}
+
+	// If Web Identity token and role were provided - use them
+	if creds, err = fetchWebIdentityTokenFromConfig(config, assumeRoleDuration); err != nil {
+		return
+	}
+	if creds != nil {
+		assumedRole = config[ConfigRole]
+		return
+	}
+
+	// Otherwise use Web Identity token file and role provided via ENV
+	if creds, err = fetchWebIdentityTokenFromFile(assumeRoleDuration); err != nil {
+		return
+	}
+	if creds != nil {
 		assumedRole = os.Getenv(roleARNEnvKey)
-	default:
-		return nil, "", errors.New("AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY required to initialize AWS credentials")
+		return
 	}
-	return creds, assumedRole, nil
+
+	err = errors.New("Missing AWS credentials, please check that either AWS access keys or web identity token are provided")
+	return
```

Comment:
Why does this function uses naked return approach, while other functions use explicit returns?

Reply:
Changed to use explicit returns, but in this particular case using naked returns looks more laconic.